examples/game/breakout.rs

```diff
@@ -1,272 +1,450 @@
-        // bound the paddle within the walls
-        translation.x = translation.x.min(380.0).max(-380.0);
+    /* Equivalently, you could spawn one brick at a time using for loops instead, at a small cost to performance
+    for row in 0..BRICK_ROWS {
+        for column in 0..BRICK_COLUMNS {
+            commands.spawn_bundle(BrickBundle::new(
+                column as f32 * (BRICK_WIDTH + BRICK_SPACING) + OFFSET_X,
+                row as f32 * (BRICK_HEIGHT + BRICK_SPACING) + OFFSET_Y,
+                &brick_material,
+            ));
+        }
     }
+    */
 }
 
-fn ball_movement_system(mut ball_query: Query<(&Ball, &mut Transform)>) {
-    if let Ok((ball, mut transform)) = ball_query.single_mut() {
-        transform.translation += ball.velocity * TIME_STEP;
+fn spawn_scoreboard(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands
+        .spawn_bundle(TextBundle {
+            text: Text {
+                sections: vec![
+                    TextSection {
+                        value: "Score: ".to_string(),
+                        style: TextStyle {
+                            font: asset_server.load(SCOREBOARD_FONT_PATH),
+                            font_size: SCORE_FONT_SIZE,
+                            color: SCOREBOARD_COLOR,
+                        },
+                    },
+                    TextSection {
+                        value: "".to_string(),
+                        style: TextStyle {
+                            font: asset_server.load(SCORE_FONT_PATH),
+                            font_size: SCORE_FONT_SIZE,
+                            color: SCORE_COLOR,
+                        },
+                    },
+                ],
+                ..Default::default()
+            },
+            style: Style {
+                position_type: PositionType::Absolute,
+                position: Rect {
+                    top: SCORE_PADDING,
+                    left: SCORE_PADDING,
+                    ..Default::default()
+                },
+                ..Default::default()
+            },
+            ..Default::default()
+        })
+        .insert(Scoreboard);
+}
+
+/// Moves everything with both a Transform and a Velocity accordingly
+fn kinematics(mut query: Query<(&mut Transform, &Velocity)>) {
+    query.for_each_mut(|(mut transform, velocity)| {
+        transform.translation.x += velocity.x * TIME_STEP;
+        transform.translation.y += velocity.y * TIME_STEP;
+    });
+}
+
+/// Turns left and right arrow key inputs to set paddle velocity
+fn paddle_input(keyboard_input: Res<Input<KeyCode>>, mut query: Query<(&Paddle, &mut Velocity)>) {
+    let (paddle, mut velocity) = query.single_mut().unwrap();
+
+    let mut direction = 0.0;
+    if keyboard_input.pressed(KeyCode::Left) {
+        direction -= 1.0;
+    }
+
+    if keyboard_input.pressed(KeyCode::Right) {
+        direction += 1.0;
     }
+
+    velocity.x = direction * paddle.speed;
```

Comment:
Per #2349, should this be scaled by delta time?

Reply:
We are moving our other objects on "fixed updates" which means they update X times within the bounds of this frame's "real delta time" and leftover accumulated time from previous frames.  Therefore if we move this on "real" delta time, we might "overstep" relative to the other things. I personally think this should also move using the fixed update delta / be a part of the fixed update system set.  Its ok to move more than once with the same input (think of this as an input sampling resolution rounding issue), but we do have both the issue of "dropped inputs" (if we had zero fixed updates this frame) and "just_pressed" not getting reset (if we had >1 fixed_updates this frame), which might result in logic repeating that wasn't intended to repeat.

I think this might be something that we need to handle better in-engine.